cpp/src/dictionary/set_keys.cu

```diff
@@ -156,7 +156,7 @@ std::vector<std::unique_ptr<column>> match_dictionaries(std::vector<dictionary_c
 {
   std::vector<column_view> keys(input.size());
   std::transform(input.begin(), input.end(), keys.begin(), [](auto& col) { return col.keys(); });
-  auto new_keys  = cudf::detail::concatenate(keys, rmm::mr::get_current_device_resource(), stream);
+  auto new_keys  = cudf::detail::concatenate(keys, stream, rmm::mr::get_current_device_resource());
```

Comment:
can remove mr here

Reply:
We can, or we can leave it explicit to make it clear we're allocating a temporary...

Reply:
That wouldn't be consistent with what we do everywhere else. If `mr` is passed along then the returned allocation is not temporary otherwise it is.

Reply:
Well, not passing it and passing `get_current_device_resource` are equivalent -- both do the right thing for temporaries, and the latter is more explicit at the point of call. But I have changed all instances, including the two pointed out.